low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -56,14 +57,39 @@ def validate(self):
             "cortx-sspl-test": None
             }
         # python 3rd party package dependency
-        pip3_3ps_packages_test = {
-            "Flask": "1.1.1"
+        pip3_packages_dep = {
+            "Flask": "1.1.1",
+            "coverage": "5.5"
             }
+        if not self.coverage_enabled:
+            pip3_packages_dep.pop("coverage")
+        # Collect installed python pkg from dependency list.
+        installed_pip3_pkgs = {}
+        for pkg in pkg_resources.working_set:
+            if pkg.project_name in pip3_packages_dep:
+                installed_pip3_pkgs[pkg.project_name] = pkg.version
+        # Validate pip3 python pkg with required version.
+        for pkg, version in pip3_packages_dep.items():
+            if pkg not in installed_pip3_pkgs or \
+               installed_pip3_pkgs[pkg] != version:
+                if pkg in installed_pip3_pkgs:
+                    cmd = f'pip3 uninstall -y {pkg}'
```

Comment:
You can use [Requirement object](https://setuptools.readthedocs.io/en/latest/pkg_resources.html#requirement-objects) from `pkg_resources`. Build a requirement string with name and version of a package and then you can use a function like this:
```
def check_package_exists(package):
    req = pkg_resources.Requirement.parse(package)
    return any(dist in req for dist in pkg_resources.working_set)
```
```
$ pip freeze | grep pytest
pytest==6.2.2

>>> check_package_exists('pytest==6.2.3')
False

>>> check_package_exists('pytest==6.2.2')
True
```

Reply:
Thats great @jjxsg , I was also checking how to use Requirement obj but failed to find any example.
This is an improvement but still we can looping through the working_set, There is one method that I see that can prevent this looping, but I failed to find how to use it can you help?
```
find(req)
Find a distribution matching req (a Requirement instance). 
If there is an active distribution for the requested project,
this returns it, as long as it meets the version requirement specified by req. 
But, if there is an active distribution for the project and it does not 
meet the req requirement, VersionConflict is raised. If there is no active 
distribution for the requested project, None is returned.
```

Reply:
That was my initial implementation, but it will require handling the exception and probably `find function` will be using similar loop behind the scenes. You can explore more, here is basic implementation:
```
def check_package_exists(package):
    result = None
    req = pkg_resources.Requirement.parse(package)
    try:
        result = pkg_resources.working_set.find(req)
    except VersionConflict:
        print("Version Conflict")
    return (result is not None)
```
```
>>> check_package_exists('pytest==6.2.2')
True
>>> check_package_exists('pytest==6.2.3')
Version Conflict
False
>>> check_package_exists('abcd==6.2.3')
False

Reply:
this looks great, Thanks 👍

Reply:
addressed 👍

Reply:
Looks fine.